Dockerfile

```diff
@@ -28,18 +27,26 @@ WORKDIR /var/lib/hypothesis
 
 # Copy packaging
 COPY h/__init__.py h/_version.py ./h/
-COPY README.rst package.json setup.* requirements.txt ./
+COPY README.rst setup.* requirements.txt ./
+COPY scripts ./scripts/
+COPY package.json gulpfile.js ./
 
 # Install application dependencies.
-RUN npm install --production \
-  && pip install --no-cache-dir -r requirements.txt \
-  && npm cache clean
+RUN pip install --no-cache-dir -r requirements.txt
 
 # Copy the rest of the application files
 COPY gunicorn.conf.py ./
 COPY conf ./conf/
 COPY h ./h/
-COPY scripts ./scripts/
+
+# Build client and static assets
+
+## Remove dependencies required only for unit testing
+RUN sed -i -r '/karma|proxyquire|phantom|mocha|chai|jscs/d' package.json
```

Comment:
@seanh - Yes, that's what @nickstenning 's just persuaded me makes more sense. Just doing that now.